File: Java/Basics/file10.java

import java.util.*;

public class file10 {

    static int count(int num) {
        int count = 0;
        while(num != 0) {
            int single = num % 10;
            count++;
            num /= 10;
        }
        return count;
    }

    static boolean is_armstrong(int num) {

        int org_num = num, sum = 0;
        int dgt_count = count(org_num);

        while(num != 0) {
            int dgt = num % 10;
             sum += (int)Math.pow(dgt, dgt_count);
             num /= 10;
        }

        return sum == org_num;

    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int num1 = sc.nextInt();
        int num2 = sc.nextInt();
        int frst_num  = 0;
        int scnd_num  = 0;

        for(int i = num1 ; i <= num2; i++) {
            if(is_armstrong(i)) {
                frst_num = i;
                 break;
            }
        }

        for(int i = num2; i > frst_num; i--) {
            if(is_armstrong(i)) {
                scnd_num = i;
                break;
            }
        }

        System.out.println(frst_num + scnd_num);
    }
}

File: C/Basics/file10.c

#include<stdio.h>
#include<math.h>

int count_digit(int dgt) {
    int sing_dgt = 0, count  = 0;
    while(dgt != 0) {
       sing_dgt = dgt % 10;
       count++;
       dgt /= 10;
    }
    return count;
}

int is_armstrong(int num) {
    

    int dgt, org_num = num, sum = 0;
    int pow_dgt = count_digit(org_num);
    while(num != 0) {
        dgt = num % 10;
        sum += (int)round(pow(dgt,pow_dgt));
        num /= 10;
    }
    if(sum ==  org_num) {
        return 1;
    }
    return 0;
}


int main()
{
    int num1, num2, flag1 = 0, flag2 = 0, first_arm = 0, scnd_arm = 0;
    scanf("%d %d", &num1, &num2);

     for(int i = num1; i <= num2; i++) {
        if(is_armstrong(i)) {
            first_arm = i;
            break;
        }
     }

     for(int i = num2; i > scnd_arm ;i--) {
        if(is_armstrong(i)) {
            scnd_arm = i;
            break;
        }
     }

     printf("%d", first_arm + scnd_arm);
}